inc/namespace.php

```diff
@@ -130,15 +132,39 @@ function get_client_side_data() : array {
 	return $data;
 }
 
+/**
+ * Adds an async attribute to the script tag output.
+ *
+ * @param string $tag The enqueued HTML script tag.
+ * @param string $handle The script handle.
+ * @return string The script tag markup.
+ */
+function async_scripts( string $tag, string $handle ) : string {
+	global $wp_scripts;
+
+	if ( ! $wp_scripts->get_data( $handle, 'async' ) || strpos( $tag, 'async' ) !== false ) {
+		return $tag;
+	}
+
+	$tag = str_replace( '></script>', ' async></script>', $tag );
+
+	return $tag;
+}
+
 /**
  * Queue up the tracker script and required configuration.
  */
 function enqueue_scripts() {
+	global $wp_scripts;
+
 	wp_enqueue_script( 'altis-analytics', plugins_url( 'build/analytics.js', __DIR__ ), [], null, false );
 	wp_add_inline_script(
 		'altis-analytics',
 		sprintf(
-			'var Altis = Altis || {}; Altis.Analytics = %s;', wp_json_encode(
+			'var Altis = Altis || {}; Altis.Analytics = %s;' .
+			'Altis.Analytics.registerAttribute = function (key, value) { Altis.Analytics._attributes[key] = value; };' .
+			'Altis.Analytics.registerMetric = function (key, value) { Altis.Analytics._metrics[key] = value; };',
```

Comment:
This is getting to be a little complex now; I wonder whether we could separate this out a bit 🤔

Reply:
I think we can refactor this whole script down some in a future issue to reduce the size of it. Functionally however this is no different than what was there previously, just these functions need to available super early.